assembly_visualization.md

```diff
@@ -0,0 +1,190 @@
+
+We want to look at:
+- the coverage regularity (ex: some repeated regions might have increased coverage)
+- the coverage at the beginning and end of scaffolds - is it good enough?
+- are they positions where pileup of reads show polymorphism?
+- ...
+
+2) to identify SNPs: some methods use reads mapping against a reference to identify and type SNPs
+
+3) Assembly polishing software such as `pilon` and `reapr` use mapped reads to identify potential improvement in assemblies. It is good to visualize what information they are using ...
+  > eg. variation in coverage, wrong read pairs orientation, discrepancy between expected insert size and actual insert size obtained from read mapping (ie. longer than expected)
+
+ - [ ] other?
+
+### 1.3 Re-map the reads to the assembly
+
+We use [bwa mem](http://bio-bwa.sourceforge.net/bwa.shtml) option from [bwa tools](http://bio-bwa.sourceforge.net/) software.
+
+You will use the trimmed-filtered reads in: `/projects/nn9305k/Bioinfo_mapping_training/bifrost_run/bbduk_trimmed` and the assembly: `/projects/nn9305k/Bioinfo_mapping_training/bifrost_run/prokka/MiSeq_Ecoli_MG1655_50x.fna`
+
+We could use the assembly output from `Pilon` **BUT** then we would have to create aliases between labels used by `Prokka` and by `Pilon` to be able to visualize the assembly with IGV. `Prokka` assembly is identical to `Pilon` assembly, only labels have been changed.
+
+In your home make a directory for today's work
+and a folder called `mapping` where you will copy the input files
+
+```bash
+# softlink the required input files in the mapping folder:
+ln -s <path>/bifrost_run/bbduk_trimmed/*.fq.gz .
+ln -s <path>/bifrost_run/pilon/MiSeq_Ecoli_MG1655_50x_pilon_spades.fasta .
+
+# we will use Bifrost with the command line
+# As bwa and samtools are used to map reads for Pilon ...
+
+qlogin --account=nn9305k --time=00:30:00 --ntasks=4 --mem-per-cpu=4G
+source activate bifrost
+
+# 1) We need to index the reference
+bwa index <reference.fasta>
+
+# 2) We map the reads (attribute the position of the reads according to the index) as PE
+bwa mem -t 4 <reference.fasta> <in1:read1.fq.gz> <in2:read2.fq.gz> \
+| samtools sort -o <out:PE_mapped_sorted.bam> -
+
+# NB: We `sort` directly the mapping by index position
+#`-` means that the output of the pipe is used as input in samtools
+
+#For unpaired reads
+bwa mem -t 4 <references.fasta> <in:U_reads.fastq.gz> \
+| samtools sort -o <out:U_mapped_sorted.bam> -
+
+# We need to merge those two files as one
+samtools merge <out:all_merged.bam> <in1:_.bam> <in2:.bam>
+
+# To be sure reads are still sorted: we resort
+samtools sort -o <out:final_all_merged.bam> <in:all_merged.bam>
+
+#NB:Some software like Pilon need that we update the index (you can do:). We will need index after
+samtools index <final_all_merged.bam>
+```
```

Comment:
Put this part into a OPTIONAL box, with explanation.

Reply:
ok